components/brave_wallet/common/fil_address.cc

```diff
@@ -119,6 +119,22 @@ FilAddress FilAddress::FromAddress(const std::string& address) {
   return FilAddress::FromPayload(payload, protocol.value(), network);
 }
 
+// static
+FilAddress FilAddress::FromBytes(const std::string& chain_id,
+                                 const std::vector<uint8_t>& bytes) {
+  if (!IsValidNetwork(chain_id)) {
+    return FilAddress();
+  }
+  if (bytes.empty()) {
+    return FilAddress();
+  }
+  mojom::FilecoinAddressProtocol protocol =
+      static_cast<mojom::FilecoinAddressProtocol>(bytes.at(0));
+  std::vector<uint8_t> payload(std::next(bytes.begin(), 1), bytes.end());
+
+  return FilAddress::FromPayload(payload, protocol, chain_id);
```

Comment:
nit: FromPayload should accept span instead of vector, so we could avoid allocating extra vectors here.